src/compiler/utilitiesPublic.ts

```diff
@@ -1986,10 +1983,6 @@ export function isUnaryExpressionWithWrite(expr: Node): expr is PrefixUnaryExpre
     }
 }
 
-/**
- * See isExpression; not for use in transforms.
- * @internal
- */
 export function isLiteralTypeLiteral(node: Node): node is NullLiteral | BooleanLiteral | LiteralExpression | PrefixUnaryExpression {

```

Comment:
I wouldn't make this version public as the type guard is actually unsound due to the call to `skipPartiallyEmittedExpressions` call (i.e., `node` could be a `PartiallyEmittedExpression` containing a `NullLiteral`, etc.).

Reply:
This test is used in visitEachChild; is that use wrong, then?

Reply:
This test probably doesn't need the `skipPartiallyEmittedExpressions` call, TBH. We only use those for actual expressions, not anywhere we produce literal types (i.e., `typeToTypeNode` or in `transformers/declarations.ts`. If we remove the call (and tests pass), then this is fine to make public.

Reply:
It passes; I think it was just cargo-culted.